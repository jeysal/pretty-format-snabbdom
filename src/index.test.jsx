// @flow
// @jsx html

import { h } from 'snabbdom';
import { html } from 'snabbdom-jsx'; // eslint-disable-line no-unused-vars
import serializer, { serialize } from '.';

// $FlowFixMe addSnapshotSerializer typings outdated, this does take an object literal
expect.addSnapshotSerializer(serializer);

test('serializes a simple node', () => {
  expect(h('div')).toMatchSnapshot();
});

test('ignores an empty node', () => {
  expect(h('')).toMatchSnapshot();
});

/*
  using selectors instead of tag names for h(selectorOrTagName)
  is not treated in a special way;
  the selector is rendered as a tag name.
  This is sufficient for snapshot testing and sometimes
  <div.cl#id />
  can even be more readable than
  <div className="cl" id="id" />
*/
test('renders a selector like a tag name', () => {
  expect(h('div.cl#id')).toMatchSnapshot();
});

test('serializes a simple JSX node', () => {
  expect(<div />).toMatchSnapshot();
});

test('serializes a JSX node and its props', () => {
  expect(
    <div
      id="id"
      className="cl"
      title="title"
      data-stuff="stuff"
      style="color: black;"
      onclick={() => true}
    />,
  ).toMatchSnapshot();
});

test('serializes a JSX node with children', () => {
  expect(
    <div>
      <div>
        <div>asdf</div>
      </div>
    </div>,
  ).toMatchSnapshot();
});

test('serializes weird text nodes', () => {
  expect(
    <div>
      {false}
      {42.1337}
      {() => true}
    </div>,
  ).toMatchSnapshot();
});

test('avoids overwriting data with data.props', () => {
  expect(h('div', { asdf: 42, props: { asdf: 1337 } })).toMatchSnapshot();
});

test('respects maxDepth', () => {
  // $FlowFixMe addSnapshotSerializer typings outdated, this does take an object literal
  expect.addSnapshotSerializer({
    ...serializer,
    serialize: (
      elem: any,
      config: Object,
      indentation: any,
      depth: any,
      refs: any,
      printer: any,
    ) =>
      serialize(
        elem,
        { ...config, maxDepth: 3 },
        indentation,
        depth,
        refs,
        printer,
      ),
  });

  expect(
    <div>
      <div>
        <div>
          <div>
            <div>asdf</div>
          </div>
        </div>
      </div>
    </div>,
  ).toMatchSnapshot();
});

test('does not interfere with serialization of other values', () => {
  expect(1).toMatchSnapshot();
  expect(true).toMatchSnapshot();
  expect('asdf').toMatchSnapshot();
  expect({}).toMatchSnapshot();
  expect({ a: 'a', b: 42 }).toMatchSnapshot();
  expect([1, 2, '3', '4']).toMatchSnapshot();
});

test('allows null and undefined children', () => {
  expect(h('div', [undefined])).toMatchSnapshot();
  expect(h('div', [null])).toMatchSnapshot();
});