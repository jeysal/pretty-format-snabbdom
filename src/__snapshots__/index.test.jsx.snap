// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allows null and undefined children 1`] = `
<div>
  undefined
</div>
`;

exports[`allows null and undefined children 2`] = `
<div>
  null
</div>
`;

exports[`avoids overwriting data with data.props 1`] = `
<div
  asdf={42}
  props={
    Object {
      "asdf": 1337,
    }
  }
/>
`;

exports[`does not interfere with serialization of other values 1`] = `1`;

exports[`does not interfere with serialization of other values 2`] = `true`;

exports[`does not interfere with serialization of other values 3`] = `"asdf"`;

exports[`does not interfere with serialization of other values 4`] = `Object {}`;

exports[`does not interfere with serialization of other values 5`] = `
Object {
  "a": "a",
  "b": 42,
}
`;

exports[`does not interfere with serialization of other values 6`] = `
Array [
  1,
  2,
  "3",
  "4",
]
`;

exports[`ignores an empty node 1`] = `
Object {
  "children": undefined,
  "data": Object {},
  "elm": undefined,
  "key": undefined,
  "sel": "",
  "text": undefined,
}
`;

exports[`renders a selector like a tag name 1`] = `<div.cl#id />`;

exports[`respects maxDepth 1`] = `
<div>
  <div>
    <div>
      <div â€¦ />
    </div>
  </div>
</div>
`;

exports[`serializes a JSX node and its props 1`] = `
<div
  style="color: black;"
  data={
    Object {
      "stuff": "stuff",
    }
  }
  id="id"
  className="cl"
  title="title"
  onclick={[Function]}
/>
`;

exports[`serializes a JSX node with children 1`] = `
<div>
  <div>
    <div>
      asdf
    </div>
  </div>
</div>
`;

exports[`serializes a simple JSX node 1`] = `<div />`;

exports[`serializes a simple node 1`] = `<div />`;

exports[`serializes weird text nodes 1`] = `
<div>
  false
  42.1337
  [Function]
</div>
`;
